himan-scripts/CB-TCU-cloud.lua

```diff
@@ -64,10 +74,11 @@ for i=1, #EL500 do
   res[i] = Missing
 
   --TCU
-  if ((Tbase[i]-Ttop[i]>TCUlimit) and (NL[i]>0) and (CIN500[i]>CINlimTCU)) then
-    res[i] = FlightLevel_(EL500[i]*100)
+  if (EL500[i] - LCL500[i] > TCUlimit) then
+    --we don't use vertical search for flight level of EL500 but calculate directly from EL500 pressure
+    res[i] = FlightLevel_(pEL500[i] * 100)
```

Comment:
This is clever 👍

Reply:
I think it is just there because smarttool function requires some value and FL650 is so high that EL500 will always be below. At least in our latitudes. And if there is 20km tall convection I think it's better to put it as it is than to turn it into missing.